test/integration/test-pages/logins-api.js

```diff
@@ -43,12 +43,10 @@ document.querySelector("#remove").addEventListener("click", () => {
 });
 document.querySelector("#update").addEventListener("click", () => {
   const log = getLogger("update");
-  const loginUpdates = {
-    guid: mockLogin.guid,
-    username: "updated",
-  };
-  browser.experiments.logins.update(loginUpdates).
-    then(log, log);
+  browser.experiments.logins.get(mockLogin.guid)
+    .then(login =>
+      browser.experiments.logins.update({ ...login, username: "updated" }))
```

Comment:
Oh, and also I _think_ this represents correctly the change in assumptions. I was already using the API this way in datastore.js, oddly enough, which made no functional difference really.

Reply:
This looks good, but I'm wondering if we should add clues about the special MetaInfo fields (including the magic [`timesUsedIncrement`](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsILoginManager#modifyLogin()) property). Would you mind also setting `timesUsedIncrement: 1` here, and also bump one of the MetaInfo timestamp fields, then adding the corresponding asserts on the test side? Maybe this'll help future contributors detect the weirdness of the underlying APIs.

Reply:
Banged my head on this for awhile and I think I've discovered this: 

If we supply both `timesUsed` and `timesUsedIncrement` in the property bag at the same time, the explicit `timesUsed` value will likely clobber whatever the result of `timesUsedIncrement` might be. 

That is, [depending on whatever the order of enumerated properties ends up being](https://searchfox.org/mozilla-central/source/toolkit/components/passwordmgr/LoginHelper.jsm#363). I haven't dug deep enough to figure out what that order is, though it seemed pretty consistently to clobber the increment.

On the bright side, I was able to update `timePasswordChanged`.

Reply:
I think it's reasonable for `update()` to actively omit `timesUsedIncrement` -- either the user of this API is properly incrementing `timesUsed` by hand, or they can use `touch()`.

thoughts? @lmorchard @6a68